### PR TITLE
[FVM] Implement program recovery for fungible tokens

### PR DESCRIPTION
**Pull request by turbolent**
Use the new program recovery feature of Cadence and add support for fungible tokens.

See https://github.com/onflow/flips/pull/283

Until the FLIP got approved, the feature is only enabled on all networks except Mainnet, for testing purposes.

### CODE REVIEW

File: fvm/environment/program_recovery.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		len(nominalType.NestedIdentifiers) == 0 &&
+		nominalType.Identifier.Identifier == name
+}
+
+const fungibleTokenTypeTotalSupplyFieldName = "totalSupply"
+const fungibleTokenVaultTypeBalanceFieldName = "balance"
+const fungibleTokenVaultTypeIdentifier = "Vault"
+
+func isFungibleTokenContract(program *ast.Program, fungibleTokenAddress common.Address) bool {
+	const contractName = "FungibleToken"
```
Comment (SupunS) [inline fvm/environment/program_recovery.go:146]: a nitpick: maybe rename this variable to `fungibleTokenContract`/ `fungibleTokenContractName`, so the code below reads a bit better. e.g: `declaresConformanceTo(contractDeclaration, fungibleTokenContract)`

**Review by SupunS — APPROVED** [review]
LGTM!

File: fvm/environment/program_recovery.go (hunk 1 above)
Comment (turbolent) [reply fvm/environment/program_recovery.go]: Good idea! Done in [6c46cee](https://github.com/onflow/flow-go/pull/6278/commits/6c46ceef0e63a9e5d6ddff8bf7bdd26cfa209298)

**Review by ramtinms — APPROVED** [review]

**Comment by turbolent** [comment]
Updated the PR: 
- The FVM feature is now gated and only enabled on all networks except Mainnet
- The replacement Fungible Token implementation is now updated to conform to the latest `FungibleToken` standard
- Added a test case for contract checking in the Cadence 1.0 migration

**Review by SupunS — APPROVED** [review]

**Pull request merged**